core/src/sumeragi/main_loop.rs

```diff
@@ -481,6 +478,78 @@ fn handle_role_agnostic_messages<F>(
             Message::ViewChangeSuggested => {
                 trace!("Received view change suggestion.");
             }
+            Message::BlockSyncUpdate(BlockSyncUpdate { block }) => {
+                let block = match block.revalidate_hashes() {
+                    Ok(block) => block,
+                    Err(err) => {
+                        warn!(?err);
+                        return;
+                    }
+                };
+                if state.latest_block_hash != block.header().previous_block_hash {
+                    warn!(
+                        expected = %state.latest_block_hash,
+                        actual = %block.header().previous_block_hash,
+                        "Latest block hash mismatch."
+                    );
+                    return;
+                }
+                if state.latest_block_height + 1 != block.header().height {
+                    warn!(
+                        expected = state.latest_block_height + 1,
+                        actual = block.header().height,
+                        "Block heights are in an inconsistent state."
```

Comment:
does this mean that their is a mismatch between actual and expected block heights? I think this message should be rewritten